### ISSUE TITLE
GridView widget problem with array query parameters

### ISSUE THREAD

**Issue by omnilight** [yiisoft/yii2#10217]
I want to use filter in grid view that allows me to select multiple values (for example `<select multiple=multiple>`). But with current implementation of the `yii.gridView::applyFilter`  Js method I can't do this, because, it does not support mulpile parameters with the same name here:
https://github.com/yiisoft/yii2/blob/master/framework/assets/yii.gridView.js#L89

I think it would be greate to rewrite this part of code to support multiple parameters with the same name

**Comment by SilverFire** [comment]
@omnilight yeah, it's a problem. Can you make a PR?

**Comment by omnilight** [comment]
#10223

**Comment by SilverFire** [comment]
https://github.com/yiisoft/yii2/commit/d3cfb87ba67a328d197d1a9e52dbb12a7bf8476b

**Comment by omnilight** [comment]
I've found small, but important bug.

I the case when you have standard `Html::activeListView` as a filter input, it really adds two inputs: one is select with the name `name[]` and second is hidden input with name `name`. Second one is passed, when no value is selected.

In the case we have selected some value, query string will be: `name=&name[]=1&name[]=2`. That's correct. Now, if you deselect all values from select, `applyFilter` function will find only hidden's input value (because of `$(settings.filterSelector).serializeArray()`). So, when later it will try to override query string values, it will not override `&name[]=1&name[]=2` part because no key `name[]` will be in the `keysInFilter` variable. So my new pull-request is fixing with behavior
